src/module.cpp

```diff
@@ -2782,10 +2779,12 @@ static bool lCompatibleTypes(llvm::Type *Ty1, llvm::Type *Ty2) {
             break;
 
         case llvm::ArrayType::PointerTyID:
+// In opaque mode pointers are compatible
```

Comment:
Is it intended behavior to have fall-though in Opaque Mode?

Reply:
I think it was assumes to `return true;` in else case of ifndef.
It also seems to affect how ISPC reports warnings about pointers to varying types in global variables.

Reply:
The intention was to have `ifndef ` before `case llvm::ArrayType::PointerTyID`, not after. But anyway I reworked it in the right way.